azurelinuxagent/ga/agent_update_handler.py

```diff
@@ -160,38 +198,142 @@ def __get_agent_family_manifests(self, goal_state):
         return agent_family_manifests[0]
 
     @staticmethod
-    def __get_requested_version(agent_family):
+    def _get_version_from_gs(agent_family):
         """
-        Get the requested version from agent family
-        Returns: Requested version if supported and available in the GS
-                 None if requested version missing or GA versioning not enabled
+        Get the version from agent family
+        Returns: version if supported and available in the GS
+                 None if version is missing
         """
-        if conf.get_enable_ga_versioning() and agent_family.is_requested_version_specified:
-            if agent_family.requested_version is not None:
-                return FlexibleVersion(agent_family.requested_version)
+        if agent_family.version is not None:
```

Comment:
Will version always be in agent_family? Should we check here that the version property exists